src/Common/Core/Impl/Imaging/IImageService.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Common.Core.Imaging {
+    /// <summary>
+    /// Provides access to images and icons
+    /// </summary>
+    public interface IImageService {
+        /// <summary>
+        /// Provides image based on one of the predefined types
+        /// </summary>
+        /// <returns>Platform-specific image source</returns>
+        object GetImage(ImageType imageType);
+
+        /// <summary>
+        /// Returns image source given name of the image moniker
+        /// such as name from http://glyphlist.azurewebsites.net/knownmonikers
+        /// when running in Visual Studio.
+        /// </summary>
+        /// <returns>Platform-specific image source</returns>
+        object GetImage(string name);
```

Comment:
`foo.png` case is represented by the next method, when we get the file path. But VS moniker is just another predefined image name, code that uses the service shouldn't care where that image comes from.